src/main/scala/rules/SymbolicExecutionRules.scala

```diff
@@ -48,6 +49,10 @@ trait SymbolicExecutionRules {
         res.counterexample = Some(finalCE)
       }
     }
+    if(Verifier.config.enableBranchconditionReporting() && v.decider.pcs.branchConditionExps.nonEmpty){

```

Comment:
The two invocations of `v.decider.pcs.branchConditionExps.nonEmpty` (this line, next line) aren't ideal, since both entail flattening the whole stack. It is probably not worth optimising here, however, since creating very many error messages (and thus calling this code) seems unlikely.

I've fiddled around with alternatives, see https://scalafiddle.io/sf/UFsXmqQ/0, but they aren't terribly exciting.

My suggestion: leave a short code comment stating the current performance penalty, but leave the code unchanged for now.